src/gpg_data.c

```diff
@@ -832,18 +858,39 @@ int gpg_apdu_get_key_data(unsigned int ref) {
 
     // encrypted part
     switch (keygpg->attributes.value[0]) {
-        case KEY_ID_RSA:  // RSA
+        case KEY_ID_RSA:
+            ksz = U2BE(G_gpg_vstate.mse_dec->attributes.value, 1) >> 3;
```

Comment:
Why we need to /8 here ?

Reply:
Legacy code done like this.
Most probably the size needed is in bytes, whereas in the descriptors it is in bits.